R/col_types.R

```diff
@@ -82,25 +82,43 @@ as.col_spec.default <- function(x) {
 }
 
 #' @export
-print.col_spec <- function(x, n = Inf, ...) {
-  cat(format.col_spec(x, n = n, ...))
+print.col_spec <- function(x, n = Inf, condense = FALSE, ...) {
+  cat(format.col_spec(x, n = n, condense = condense, ...))
 
   invisible(x)
 }
 
+condense_spec <- function(x) {
+  types <- vapply(x$cols, function(xx) class(xx)[[1]], character(1))
+  counts <- table(types)
+  most_common <- names(counts)[counts == max(counts)][[1]]
+
+  x$default <- x$cols[types == most_common][[1]]
+  x$cols <- x$cols[types != most_common]
+  x
+}
+
 #' @export
-format.col_spec <- function(x, n = Inf, ...) {
+format.col_spec <- function(x, n = Inf, condense = FALSE, ...) {
```

Comment:
I think it should be condensed automatically in the auto-printing in `read_delimited()`